docs/api.md

```diff
@@ -6,6 +6,202 @@ challenges presented by JavaScript and web browsers.
+that, when resolved, loads into the notebook the save file `fileName`. For each
+notebook can find a list of uploaded files in the [upload modal](#link-required)
+or on your notebook's [Revisions Page](#needs-link). You can also access any
+uploaded file through the fetch chunk, [following this
+pattern](https://iodide-project.github.io/docs/workflows/#uploading-data-to-an-iodide-notebook).
+For most use cases using [fetch
+chunks](https://iodide-project.github.io/docs/jsmd/#fetch-chunks-fetch) is
+preferable and more straightforward. In more dynamic programmatic cases,
+however, `iodide.file.load` can provide more nuanced workflows.
+
+`fileName` is the name of the file uploaded to the Iodide server. 
+
+`fileType` is the file type to handle. These are the same as the file types in
+[fetch chunks](https://iodide-project.github.io/docs/jsmd/#fetch-chunks-fetch):
+`js` (load this file as javascript, eg a library), `css` (load this file as a
+stylesheet, applying the styles to the report), `json` (load this file as json
+and parse into a javascript object), `text` (load this file as text), `blob`
+(load this file as binary).
+
+`variableName` (required for `json`, `text`, and `blob` file types, otherwise
+not applicable): the variable name in which to load the data, available in the
+browser `window` namespace.
+
+
+#### `iodide.file.load` patterns
+
+```javascript
+
+// load a csv
+iodide.file.load('cached-data.csv', 'text', 'cachedData').then(() => {
+  cachedData = d3.csvParsed(cachedData)
+})
+
+// load a video
+iodide.file.load('gritty.mp4', 'blob', 'gritty').then(() => {
+  // load this gritty video into a <video /> tag in the report.
+  var urlCreator = window.URL || window.webkitURL;
+  var imageUrl = urlCreator.createObjectURL(gritty);
+  document.querySelector("#gritty").src = imageUrl;
+})
+
+// load a style sheet and apply the style
+iodide.file.load('corporate-report-style.css', 'css')
+
+// load some javascript helpers
+iodide.file.load('analysis-helpers.js', 'js')
+
+```
+
+
+
+### `iodide.file.delete(fileName[, deleteOptions])`
+
+Deletes the file specified by `fileName`. Returns a
+[Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises)
+that when resolved, denotes that the file was deleted on the server.
+
+`deleteOptions` is an object with the following keys:
+
+- `throwIfDoesNotExist` (optional, default `true`): will throw an error and halt
+  the evaluation queue if `true`.
+
+
+
+### `iodide.file.list()`
+
+Returns an Array of file names available to the current notebook.
+
+
+#### `iodide.file.list` patterns
+
+
+```javascript
+// delete the various pngs originally saved to this notebook
+iodide.file.list()
+  .filter(filename => !filename.includes('.png'))
+  .forEach(iodide.file.delete)
+```
+
+
+### `iodide.file.exists(fileName)`
+
+Returns `true` if the file `fileName` is available to the notebook, and `false`
```

Comment:
Would this need to be async, and thus return a `Promise`?

Reply:
It would not need to be async because we will (soon) have all the file metadata brought into the notebook from the server upon load.

Reply:
Oh, that's very cool, actually.